src/allowlist/AllowList.sol

```diff
@@ -23,9 +23,13 @@ contract AllowList is IAllowListV2, Ownable2StepUpgradeable {
     mapping(address => EntityId) public addressEntityIds;
 
     /// @notice A record of permissions for each entityId determining if they are allowed.
-    mapping(EntityId => mapping(string => bool)) fundPermissionsByEntityId;
+    mapping(EntityId => mapping(string fundSymbol => bool permission)) public fundPermissionsByEntityId;
 
-    /// Future allow lists could be added here
+    /// @notice TODO
+    mapping(address contractAddress => uint256 numberOfFunds) public contractPermissionsForFunds;
+
+    /// @notice Contract address permissions, mutually exclusive with entityId permissions
+    mapping(address => mapping(string fundSymbol => bool permission)) public contractPermissions;
```

Comment:
nit: an entity address could be a contract too. maybe a name like"protocolPermissions" or "nonEntityPermissions" something would be more precise? 

speaking of, i dont think we'd ever want a non-contract in here right? we might want to contract size != 0 as a double check inside `setContractAddressPermission`.